### DISCUSSION TITLE
Github Actions

### DISCUSSION THREAD

**Discussion by hariombalhara** [johnlindquist/kit#409] [Share]
[Github Actions Script](https://gist.github.com/hariombalhara/969bd98ec14006d5b8b60fb51d018aa2)

I have my blog using Notion API and whenever I want to publish a post to my blog, I trigger a GitHub Workflow that does the job.

Earlier, I used to do it using a bookmark for the workflow and 1-2 additional clicks in Github UI.

**Comment by johnlindquist** [comment]
Ooooh, nice! I hadn't thought about trigger a GitHub action from Script Kit. Great idea. I'm going to ping Mr. Octokit to see if he has any recommendations/tips for Octokit + github actions

**Reply by gr2m** [reply]
### 👋😆 

so yeah you could make the script a bit more resilient and easier to read by using [`@octokit/core`](https://github.com/octokit/core.js), but the current script uses no dependencies right now, I don't know how much friction it would cause to add a dependency?

One thing I'd recommend to change is the name of the `actions` and `actionName` variables to `workflows` and `workflowName`. At least to make that would make it more clear. A workflow is a set of jobs which in turn is a set of steps, each step is an action. You are loading workflows, not actions

**Reply by hariombalhara** [reply]
> One thing I'd recommend to change is the name of the actions and actionName variables to workflows and workflowName. At least to make that would make it more clear. A workflow is a set of jobs which in turn is a set of steps, each step is an action. You are loading workflows, not actions

Thanks, @gr2m. You are correct. Let me fix it quickly.

**Reply by hariombalhara** [reply]
Done the changes :)

Had never heard of Octokit. Would use it if the script starts getting more complex.
